RxSwift/ObservableType.swift

```diff
@@ -8,8 +8,10 @@
 
 /// Represents a push style sequence.
 public protocol ObservableType : ObservableConvertibleType {
+    #if !swift(>=3.2)
     /// Type of elements in sequence.
     associatedtype E
```

Comment:
```
/Users/alan/repositories/github/RxSwift/RxSwift/ObservableType.swift:12:20: warning: redeclaration of associated type 'E' from protocol 'ObservableConvertibleType' is better expressed as a 'where' clause on the protocol
    associatedtype E
    ~~~~~~~~~~~~~~~^
    
/Users/alan/repositories/github/RxSwift/RxSwift/ObservableConvertibleType.swift:12:20: note: 'E' declared here
    associatedtype E
                   ^
```

This one was interesting because the compiler hint would just insert a `where {` with no symbol.